src/core_ocean/shared/mpas_ocn_diagnostics.F

```diff
@@ -1513,15 +1513,12 @@ subroutine ocn_compute_KPP_input_fields(statePool, forcingPool, meshPool, diagno
         deltaVelocitySquared = 0.0_RKIND
         do i = 1, nEdgesOnCell(iCell)
           iEdge = edgesOnCell(i, iCell)
-          factor = 0.5_RKIND * dcEdge(iEdge) * dvEdge(iEdge) * invAreaCell
           delU2 =  (surfaceStress(iEdge))**2
-          deltaVelocitySquared = deltaVelocitySquared + factor * delU2
+          deltaVelocitySquared = deltaVelocitySquared + edgeAreaFractionOfCell(i,iCell) * delU2
         enddo
-        surfaceStressMagnitude(iCell) = sqrt(deltaVelocitySquared)
-
-       ! compute surface friction velocity
-         surfaceFrictionVelocity(iCell) = sqrt(surfaceStressMagnitude(iCell) / rho_sw)
-
+       ! NOTE that the factor of 2 is from averaging dot products to cell centers on a C-grid
+        surfaceStressMagnitude(iCell) = sqrt(2.0_RKIND * deltaVelocitySquared)
+        surfaceFrictionVelocity(iCell) = sqrt(surfaceStressMagnitude(iCell) / rho_sw)
```

Comment:
@vanroekel Looking at units:
```
surfaceStressMagnitude N m^{-2} = kg m s^{-2} m^{-2} = kg m^{-1} s^{-1}
surfaceFrictionVelocity = sqrt(surfaceStressMagnitude(iCell) / rho_sw)
 sqrt( kg m^{-1} s^{-2} / (kg/m^3) ) = sqrt( m^2 / s^2 )  = m/s
```
Yes, that is right and I agree with your formulation. I was tricked by the summation variable deltaVelocitySquared, which has units of `(N m^{-2})^2)`, not `( m/s)^2`. I'll change that to `sumSurfaceStressSquared` for clarity.